contextily/plotting.py

```diff
@@ -0,0 +1,34 @@
+"""Tools to plot basemaps"""
+
+from . import tile_providers as sources
+from .tile import _calculate_zoom, bounds2img, _sm2ll
+
+def add_basemap(ax, zoom='auto', url=sources.ST_TERRAIN, 
+		interpolation='bilinear', attribution_text = '', 
+                **extra_imshow_args):
+    '''
+    Add a basemap to `ax`
+
+    NOTE: `ax` needs to be in Web Mercator
+    '''
+    # Extent
+    left, right = ax.get_xlim()
+    bottom, top = ax.get_ylim()
+    # If web source
+    if url[:4] == 'http':
+        # Zoom
+        if (type(zoom) == str) and (zoom.lower() == 'auto'):
```

Comment:
`isinstance(zoom, str)` is better here

(and personally I would not do the `lower`, IMO no reason to give the user this flexibility)

Reply:
Addressed in 5a37032950c8d9df5a0ecd041dd8693edcb89076, although I've left `lower` as I think it's a way to be gentler with people who always forget whether it's `auto`, `Auto`, `AUTO`, etc.